zkvm/src/blockchain/state.rs

```diff
@@ -0,0 +1,89 @@
+use bulletproofs::BulletproofGens;
+use musig::VerificationKey;
+use std::collections::{HashSet, VecDeque};
+
+use super::block::{Block, BlockHeader, BlockID};
+use super::errors::BlockchainError;
+use crate::{Entry, TxLog, VMError, UTXO};
+
+#[derive(Clone)]
+pub struct BlockchainState {
+    pub initial: BlockHeader,
+    pub tip: BlockHeader,
+    pub utxos: HashSet<UTXO>,
+    pub nonces: VecDeque<([u8; 32], u64)>, // xxx need fast lookup by anchor
```

Comment:
Maybe a [Map](https://doc.rust-lang.org/std/iter/struct.Map.html)?

Reply:
Sure, but I think that's for a followup PR. The code that needs to do the lookup doesn't exist yet, and remember, we're just trying to get this landed so everyone else can start filling in parts of it.

Reply:
Makes sense to me!